jina/helper.py

```diff
@@ -156,13 +156,13 @@ def batch_iterator(data: Union[Iterator[Any], List[Any], np.ndarray], batch_size
                 yield tuple(sl)
             else:
                 yield data[tuple(sl)]
-    elif hasattr(data, '__len__'):
+    elif isinstance(data, Sequence):
```

Comment:
Since we are here just changing type, let's avoid doing logical changes, what is the error identified here?

A `dict` for instance has attribute `length`, is an `Iterable` but not a `Sequence`, so according to new typing it is an "expected" input and we are changing its behavior here.

Reply:
The error here is that `dict`s don't support slicing. Thus the line `yield data[_:_ + batch_size]` will throw a `TypeError` for a `dict`.